Terraform LSP failed to init

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The TF LSP fails to initialise when opening a `.tf` file both with and without TF in the PATH.

### Environment

Zed: v0.145.0 (Zed Nightly e8d674dc04fdd3f74946014ef92e76fd9b78ba9c)
OS: macOS 14.5.0
Memory: 18 GiB
Architecture: aarch64

Terraform: v0.0.3

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
[INFO] retrying installation of language server "terraform-ls" in 1s
[INFO] About to spawn test binary
[WARN] test binary failed to launch
[WARN] test binary check failed
[INFO] beginning to reinstall server
[INFO] deleting server container
[ERROR] server container removal

Caused by:
    No such file or directory (os error 2)
[INFO] starting language server "terraform-ls", path: "/<user-path>/<project-path>", id: 6
[INFO] starting language server. binary path: "/<user-path>/Library/Application Support/Zed/extensions/work/terraform/terraform-ls-v0.34.0/terraform-ls", working directory: "/<user-path>/<project-path>", args: ["serve"]
[ERROR] failed to start language server "terraform-ls": failed to spawn command. path: "/<user-path>/Library/Application Support/Zed/extensions/work/terraform/terraform-ls-v0.34.0/terraform-ls", working directory: "/<user-path>/<project-path>", args: ["serve"]
[ERROR] server stderr: Some("")
[ERROR] Hit 4 reinstallation attempts for "terraform-ls"
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/extensions/terraform/src/terraform.rs
+++ b/extensions/terraform/src/terraform.rs
@@ -62,6 +62,8 @@ impl TerraformExtension {
             zed::download_file(&download_url, &version_dir, zed::DownloadedFileType::Zip)
                 .map_err(|e| format!("failed to download file: {e}"))?;
 
+            zed::make_file_executable(&binary_path)?;
+
             let entries =
                 fs::read_dir(".").map_err(|e| format!("failed to list working directory {e}"))?;
             for entry in entries {

